### PR TITLE
Table: Update filter UI to be less confusing

### LINKED ISSUE #90800: Table: Filter != not working as expected
When applying the `!=` filter to a table column the value that should be excluded is instead being exclusively included - similar to what we would expect the `=` functionality to be.

Screenshot 2024-07-22 at 4 12 16 PM

[debug-Automobiles Data Table-2024-07-22 16_12_23.json.txt](https://github.com/user-attachments/files/16340532/debug-Automobiles.Data.Table-2024-07-22.16_12_23.json.txt)

### PR DESCRIPTION
**Pull request by codeincarnate**
**What is this feature?**

This PR updates the filtering UI on the table to be less confusing. https://github.com/grafana/grafana/issues/90800 has outlined a common complaint, that it seems like the interface which will search through filterable values is actually a filter in and of itself. This PR addresses this by doing the following:

1.  Improving labels to make the functionality clear in the UI
2. Re-organizing the search interface to make more clear it searches values for filtering
3. Bring case insensitive search control alongside search interface
4. Moving "Select All" to a more reasonable place
5. Tightening up the overall UI

**Note**: Issue [Docs: update Table docs to reflect new filter UI #94432](https://github.com/grafana/grafana/issues/94432) must be addressed before this PR can be merged.

### Before

Screenshot 2024-09-26 at 12 52 56 PM

### After

Screenshot 2024-09-26 at 12 48 09 PM

**Why do we need this feature?**

To make the table filtering UI less confusing.

**Who is this feature for?**

Users of the table

**Which issue(s) does this PR fix?**:

Fixes #90800

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

**Comment by codeincarnate** [comment]
@staton-hyse11 would be good to review this once you're back in 😄

**Review by Develer — APPROVED** [review]
LGTM! 👍 

I have an urge to add more filters in this case (e.g. less than 80) 😄 
Screenshot 2024-09-30 at 13 35 52

But I'm sure somewhere in the backlog we have a follow-up about extended filters 😀

File: packages/grafana-ui/src/components/Table/FilterPopup.tsx [hunk 1]
```diff
@@ -95,7 +79,7 @@ export const FilterPopup = ({
           <Stack gap={3}>
```
Comment (Develer) [inline packages/grafana-ui/src/components/Table/FilterPopup.tsx:79]:
```suggestion
          <Stack justifyContent={'space-between'}>
```

I think it looks better 😄 

Before:
Screenshot 2024-09-30 at 14 07 44

After:
Screenshot 2024-09-30 at 14 07 50

**Comment by Develer** [comment]
It's not clear to me what's the final design decision [here](https://www.figma.com/design/hIw3jiqBqWNcKA3qufUiWb/Table-filter?node-id=0-1&node-type=canvas&t=ZwZE6q8m2RU2PZvW-0).

I'm not sure about removing the `contain`, `greater`, `less`, etc. dropdown, I find it very useful. But the filter target should be different.

The main source of confusion is that we are filtering the filter instead of using the filter to filter the actual table values. So my suggestion is to remove the values block from the filter pop-up window, this will remove confusion about what user filter. And this approach is adopted by multiple other products and it's gonna be more clear for users by default.

Something like this:
Screenshot 2024-10-14 at 14 19 41

cc @staton-hyse11

**Pull request closed without merging**